rts/Game/GameHelper.cpp

```diff
@@ -934,10 +934,13 @@ void CGameHelper::BuggerOffRectangle(const float3& mins, const float3& maxs, boo
 	const int allyTeamId = teamHandler.AllyTeam(teamId);
 
 	for (CUnit* u : *qfQuery.units) {
-		if (u->moveDef == nullptr) { continue; }
+		if (u->unitDef == nullptr || u->unitDef->IsImmobileUnit()) { continue; }
 
-		const float footPrintX = (u->moveDef->xsizeh+1)*SQUARE_SIZE;
-		const float footPrintZ = (u->moveDef->zsizeh+1)*SQUARE_SIZE;
+		// apparently air units may not have a move def.
+		const bool useMoveDef = (u->moveDef != nullptr);
+
+		const float footPrintX = ((useMoveDef ? u->moveDef->xsizeh : u->xsize/2) + 1) * SQUARE_SIZE;
+		const float footPrintZ = ((useMoveDef ? u->moveDef->zsizeh : u->zsize/2) + 1) * SQUARE_SIZE;
```

Comment:
```suggestion
		const float footPrintX = ((useMoveDef ? u->moveDef->xsizeh : u->xsize/SPRING_FOOTPRINT_SCALE) + 1) * SQUARE_SIZE;
		const float footPrintZ = ((useMoveDef ? u->moveDef->zsizeh : u->zsize/SPRING_FOOTPRINT_SCALE) + 1) * SQUARE_SIZE;
```
See
https://github.com/beyond-all-reason/spring/blob/8d1ecf0dbd3702ed6b02a74b31b3ee14cd1f66b0/rts/Sim/Misc/GlobalConstants.h#L17

Reply:
No, I really want half the unit size here, which isn't provided as a property like it is for the moveDef (notice the xsizeh)

Reply:
I see.